Resources/public/js/views/services/ez-roleserversideviewservice.js

```diff
@@ -0,0 +1,281 @@
+                //    "values": {
+                //        "ref": [{
+                //            "_href": "/api/ezp/v2/content/sections/1",
+                //            "_media-type": "application/vnd.ez.api.Section+json"
+                //        }, {
+                //            "_href": "/api/ezp/v2/content/sections/4",
+                //            "_media-type": "application/vnd.ez.api.Section+json"
+                //        }]
+                //    }
+                //};
+                //var limitation = { // DUMMY SUBTREE DATA
+                //    "_identifier": "Subtree",
+                //    "values": {
+                //        "ref": [{
+                //            "_href": "/api/ezp/v2/content/locations/1/5/14",
+                //            "_media-type": "application/vnd.ez.api.Location+json"
+                //        }]
+                //    }
+                //};
+
+                // Convert hrefs to IDs, required by API
+                if (limitation) {
+                    Y.Array.each(limitation.values.ref, function (ref, i, refs) {
+                        if (limitation._identifier == "Section") {
+                            refs[i]._href = /\d+$/.exec(ref._href)[0];
+                        } else if (limitation._identifier == "Subtree") {
+                            refs[i]._href = /(\/\d+)+$/.exec(ref._href)[0];
+                        }
+                    });
+                }
+
+                roleAssignInputStruct = userService.newRoleAssignInputStruct(role, limitation);
+
+                if (contentTypeIdentifier === 'user') {
+                    service._assignRoleToUser(roleAssignInputStruct, struct.content, end);
+                } else if (contentTypeIdentifier === 'user_group') {
+                    service._assignRoleToUserGroup(roleAssignInputStruct, struct.content, end);
+                } else {
+                    service._error('The selected content "' + struct.content.get('name') + '" is not a user or user group.');
+                }
+            });
+        },
+
+        /**
+         * Assign the role to a user
+         *
+         * @method _assignRoleToUser
+         * @protected
+         * @param {RoleAssignInputStruct} roleAssignInputStruct
+         * @param {Content} content
+         * @param {Function} end
+         */
+        _assignRoleToUser: function (roleAssignInputStruct, content, end) {
+            var userService = this.get('capi').getUserService(),
+                discoveryService = this.get('capi').getDiscoveryService(),
+                service = this,
+                userIdStr;
+
+            discoveryService.getInfoObject('users', function (error, usersInfoObject) {
+                if (error) {
+                    service._error('getInfoObject for "users" failed: ' + error.message);
+                    return;
+                }
+
+                // We need a user ID, e.g.: /api/ezp/v2/user/users/10
```

Comment:
here as well you should add a TODO with a link to JIRA saying that the UDW should allow to browse the users/user groups only and should then provide the User or User Group instead of the Content, Location and cie.